lib/cb/models/implementations/saved_search.rb

```diff
@@ -36,18 +36,19 @@ def initialize(args = {})
         @search_parameters          = SearchParameters.new(args['SavedSearchParameters'] || {})
       end
 
-      def create_to_xml
-        <<-eos
-          <Request>
-            <HostSite>#{host_site}</HostSite>
-            <Cobrand>#{cobrand}</Cobrand>
-            <SearchName>#{search_name}</SearchName>
-            #{search_parameters.to_xml}
-            <IsDailyEmail>#{is_daily_email.to_s.upcase}</IsDailyEmail>
-            <ExternalUserID>#{external_user_id}</ExternalUserID>
-            <DeveloperKey>#{Cb.configuration.dev_key}</DeveloperKey>
-          </Request>
-        eos
+      def create_to_json
+        hash = {
+          'SiteID' => site_id,
+          'Cobrand' => cobrand,
+          'EmailDeliveryDay' => email_delivery_day,
+          'IsDailyEmail' => is_daily_email.to_s.upcase,
+          'userOAuthToken' => user_oauth_token,
+          'HostSite' => host_site,
+          'SearchName' => search_name,
+          'SavedSearchParameters' => search_parameters.to_hash
+        }
+        hash['EmailDeliveryDay'] = email_delivery_day unless is_daily_email
```

Comment:
So the above is correct. Reason is that on MX side EmailDeliveryDay and IsDailyEmail are each a half proxy for the IsDailyEmail property. Code looks at both and decides what the actual value should be. So, if EmailDeliveryDay exists and has a value, it overrides the IsDailyEmail field. This would be bad if IsDailyEmail is set to true because it would end up being set to false on MX side and saved that way.